src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -587,6 +587,15 @@ namespace winrt::TerminalApp::implementation
         case CommandPaletteMode::TabSwitchMode:
             return _tabSwitcherMode == TabSwitcherMode::MostRecentlyUsed ? _mruTabActions : _tabActions;
         case CommandPaletteMode::CommandlineMode:
+            _commandLineHistory.Clear();
```

Comment:
If you're going to clear it each time anyways, does it make sense to just get rid of the member and use the ApplicationState directly? Feels a little weird to create/destroy two vectors each time the command palette is summoned/used.

Reply:
Yeah, considered this and hesitating. The idea is to avoid allocation of a vector of filtered commands, but probably not worth it. I will fix it.